src/features/takeOtc/takeOtcSlice.ts

```diff
@@ -34,6 +36,15 @@ export const takeOtcSlice = createSlice({
         status: action.payload,
       };
     },
+    setCancelInProgress: (
+      state,
+      action: PayloadAction<TakeOtcState["cancelInProgress"]>
+    ): TakeOtcState => {
+      return {
+        ...state,
+        cancelInProgress: action.payload,
+      };
+    },
```

Comment:
Make a `useCancelPending` hook to know if an order is pending. Use `useDepositPending` as example.